lib/asset/assetmetadata.js

```diff
@@ -38,7 +38,7 @@ class AssetMetadata {
                 throw new IllegalArgumentError("'contentType' must be a string", contentType);
             }
         }
-        if (!Number.isFinite(contentLength) || (contentLength < 0)) {
+        if (contentLength && !Number.isFinite(contentLength) || (contentLength < 0)) {
```

Comment:
this was the key:
- make sure it does not fail in the AssetMetadata constructor if `contentLength` is missing (this is because the head request is called after initializing the metadata